cdap-docs/developers-manual/source/security/index.rst

```diff
@@ -73,3 +73,24 @@ If the standard authentication mechanisms are not sufficient, you can provide a
 :doc:`Authorization Extensions: <authorization-extensions>` Authorization backends for CDAP
 are implemented as extensions. Extensions run in their own, isolated classloader so that
 there are no conflicts with the system classloader of CDAP Master.
+
+
+.. impersonation-start
+
+.. rubric:: Impersonation
+
+Impersonation allows users to run programs and access datasets, streams, and other
+resources as pre-configured users (a *principal*). Currently, CDAP supports configuring
+impersonation at a namespace and at an application level, with application level
+configuration having a higher precedence than namespace level.
+
+If impersonation is enabled, and you don't specify a principal for an application, stream,
+or dataset, then the namespace owner's principal is used. If there is no namespace owner
+or you are using the default namespace, then the default principal is used (as set by the
```

Comment:
This text is inserted elsewhere, so the other places where it is used are exactly the same:

- Building Blocks > Streams
- Building Blocks > Datasets